lib/publiccloud/provider.pm

```diff
@@ -553,14 +553,17 @@ sub terraform_apply {
     my $tf_log = get_var("TERRAFORM_LOG", "");
 
     # The $terraform_timeout must higher than $terraform_vm_create_timeout (See also var.vm_create_timeout in *.tf file)
-    my $tf_apply_output = script_output("TF_LOG=$tf_log terraform apply -no-color -input=false myplan", timeout => $terraform_timeout, proceed_on_failure => 1);
-    my $exit_code = script_output('echo $?');
+    my $ret = script_run("set -o pipefail; TF_LOG=$tf_log terraform apply -no-color -input=false myplan 2>&1 | tee tf_apply_output", timeout => $terraform_timeout);
+    my $tf_apply_output = script_output('cat tf_apply_output', proceed_on_failure => 1);
     $self->terraform_applied(1);    # Must happen here to prevent resource leakage
 
+    record_info("TFM apply output", $tf_apply_output);
```

Comment:
such long text looks not very nice ( please check you VR only "TFM" is visible ) it is preferable to use single word for first parameter of record_info